client/js/table-view.js

```diff
@@ -41,55 +43,178 @@ class TableView {
 
   renderTableHeader() {
     removeChildren(this.headerRowEl);
+
+    //add blank column header for row numbers colummn
+    let th = createTH();
+    th.className = 'row-numbers';
+    this.headerRowEl.appendChild(th);
+
     getLetterRange('A', this.model.numCols)
       .map(colLable => createTH(colLable))
-      .forEach(th => this.headerRowEl.appendChild(th));
+      .forEach(th => {
+        th.className = 'normal-cell';
+        this.headerRowEl.appendChild(th)
+      });
+
+    th = createTH('+');
+    th.className = 'row-numbers';
+    this.headerRowEl.appendChild(th);
   }
   
   isCurrentCell(col, row) {
     return this.currentCellLocation.col === col &&
            this.currentCellLocation.row === row;
   }
 
-  renderTableBody() {
+  renderTableBody(selectedColumn, selectedRow) {
     const fragment =  document.createDocumentFragment();
-    for (let row = 0; row < this.model.numRows; row += 1) {
+    for (let row = 0; row < this.model.numRows ; row += 1) {
       const tr = createTR();
-      for (let col = 0; col < this.model.numCols; col += 1) {
+      if (row === this.model.numRows -1) {
+        tr.className = 'sum-row';
+      }
+      if (row === selectedRow) {
+        tr.className = 'selected-row';
+      }
+      for (let col = 0; col < this.model.numCols + 1; col += 1) {
         const position = {col: col, row: row};
         const value = this.model.getValue(position);
         const td = createTD(value);
-
-        if (this.isCurrentCell(col, row)) {
+        
+        if (col === 0) {
+          td.className = 'row-numbers';
+        }
+        if (col === selectedColumn) {
+          td.className = 'selected-column';
+        }
+        if (this.isCurrentCell(col, row) && !selectedColumn && !selectedRow) {
           td.className = 'current-cell';
         }
         tr.appendChild(td);
       }
       fragment.appendChild(tr);
     }
+    // add '+' cell in new row for 'add a row' functionality
+    const tr = createTR();
+    this.model.setValue({col: 0, row: this.model.numRows + 1}, '+');
+    const td = createTD(this.model.getValue({col: 0, row: this.model.numRows + 1}));
+    td.className = 'row-numbers';
+    tr.appendChild(td);
+    fragment.appendChild(tr);
+
     removeChildren(this.sheetBodyEl);
     this.sheetBodyEl.appendChild(fragment);
+
+  }
+  
+  renderColumnSum(col) {
+    let result = 0;
+    for(let i = 0; i < this.model.numRows - 1; i += 1) {
```

Comment:
Replace all += 1 instances with "++" (increments by one). Likewise, a -= 1 can be replaced with "--" (decrements by 1).